### PR TITLE
*: pre-allocate memory for slices

### PR DESCRIPTION
**Pull request by jianzhiyao**
### What problem does this PR solve?

For Issue Number: https://github.com/pingcap/tidb/issues/5069

Problem Summary:
- branch: master
- hash: 24a53c392f0ea39ee4fcf805f45014d0498a3988
```
> prealloc ./...
cmd\importer\db.go:119 Consider preallocating values
executor\sample.go:211 Consider preallocating ranges
executor\distsql.go:745 Consider preallocating handleOffset
executor\infoschema_reader.go:1062 Consider preallocating rows
executor\infoschema_reader.go:1073 Consider preallocating rows
executor\infoschema_reader.go:1088 Consider preallocating rows
executor\infoschema_reader.go:1191 Consider preallocating rows
executor\infoschema_reader.go:1651 Consider preallocating rows
executor\infoschema_reader.go:2131 Consider preallocating columnNames
executor\infoschema_reader.go:2172 Consider preallocating rows
executor\inspection_result.go:711 Consider preallocating results
executor\inspection_result.go:711 Consider preallocating results
executor\memtable_reader.go:265 Consider preallocating results
executor\memtable_reader.go:332 Consider preallocating results
executor\memtable_reader.go:398 Consider preallocating serversInfo
executor\memtable_reader.go:497 Consider preallocating levels
executor\memtable_reader.go:545 Consider preallocating results
executor\analyze_test.go:332 Consider preallocating colsInfo
executor\inspection_result_test.go:482 Consider preallocating servers
executor\memtable_reader_test.go:882 Consider preallocating servers
infoschema\tables.go:1455 Consider preallocating servers
infoschema\tables.go:1506 Consider preallocating servers
infoschema\tables.go:1589 Consider preallocating servers
infoschema\perfschema\tables.go:379 Consider preallocating results
planner\core\exhaust_physical_plans.go:2039 Consider preallocating columns
planner\core\memtable_predicate_extractor.go:80 Consider preallocating results
planner\core\rule_join_reorder_dp.go:210 Consider preallocating otherConds
planner\core\point_get_plan.go:963 Consider preallocating visitInfos
privilege\privileges\cache.go:1087 Consider preallocating gs
server\conn.go:1622 Consider preallocating rowKeys
store\copr\batch_coprocessor.go:324 Consider preallocating regionInfos
store\mockstore\unistore\tikv\mvcc_test.go:1504 Consider preallocating keys
store\mockstore\unistore\tikv\mvcc_test.go:1505 Consider preallocating vals
store\tikv\tests\lock_test.go:484 Consider preallocating locks
table\tables\partition.go:355 Consider preallocating remainPgs
table\tables\tables.go:547 Consider preallocating pkColIds
table\tables\tables.go:575 Consider preallocating pkCols
util\misc.go:260 Consider preallocating vs
util\processinfo.go:149 Consider preallocating l
util\ranger\detacher.go:33 Consider preallocating accessConditions
util\rowcodec\rowcodec_test.go:743 Consider preallocating cols
```

### What is changed and how it works?

pre-allocate memory for slices

### Check List 

Tests 

- No code

Side effects

- Performance regression
    - Consumes less CPU
    - Consumes less MEM

### Release note 

- pre-allocate memory for slices

### CODE REVIEW

**Comment by bb7133** [comment]
Thanks for your contribution!

**Comment by tangenta** [comment]
@jianzhiyao How did you find these cases? Did you use any linter or other tool?

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/cznic/mathutil v0.0.0-20181122101859-297441e03548
 	github.com/cznic/sortutil v0.0.0-20181122101858-f5f958428db8
 	github.com/danjacques/gofslock v0.0.0-20191023191349-0a45f885bc37
-	github.com/dgraph-io/ristretto v0.0.1
+	github.com/dgraph-io/ristretto v0.0.3
```
Comment (tangenta) [inline go.mod:16]: Why do we need to update this library?

**Comment by jianzhiyao** [comment]
> @jianzhiyao How did you find these cases? Did you use any linter or other tool?
My comment is updated, plz see that on the top again

**Comment by tangenta** [comment]
Ok, thanks! I guess this PR only fixes a part of the issues, right? Maybe we can integrate [prealloc](https://github.com/alexkohler/prealloc) to the CI later.

**Review by tangenta — COMMENTED** [review]
/lgtm

**Review by tangenta — APPROVED** [review]

**Comment by jianzhiyao** [comment]
> Ok, thanks! I guess this PR only fixes a part of the issues, right? Maybe we can integrate [prealloc](https://github.com/alexkohler/prealloc) to the CI later.

Yes, I just fix a part of it. Some case's slice length can not be sure. :-)

**Comment by jianzhiyao** [comment]
image
@tangenta Seems something is wrong?

**Pull request merged**